pyrsp/rsp.py

```diff
@@ -407,7 +407,12 @@ def run(self, start=None, setpc=True):
         while kind in (b'T', b'S') and sig == 5:
             # Update current thread for a breakpoint handler.
             event = stop_event(data)
-            self.thread = event[b"thread"]
+            # If server does not specify a thread explicitly then assume that
+            # current thread has not been changed.
+            # XXX: There is no statement found in the protocol specification
+            # about that aspect.
+            if b"thread" in event:
+                self.thread = event[b"thread"]
```

Comment:
shouldn't we unset self.thread in the else branch?

Reply:
`self._thread`, actually. `self.thread` is a `property` wrapper. I think, we should not unset it. Because, always there is a current branch. There are two options:
1. left all as is;
2. request current thread explicitly from the server.

2-nd option results in extra overhead. Hence, we mast have a reason to do that.

Reply:
I can write a test that launches a multi-threaded ARM application in Qemu-user. But, it requires fix for #12.

Reply:
ah, so if there is no thread in event, then it means it's the default thread? then i guess we should leave it alone, you're right.

Reply:
Actually, there might be no current thread in `RSP._thread` because user may specify either "any" or "all" threads before "vCont" ("c").

> Several packets and replies include a thread-id field to identify a thread. Normally these are positive numbers with a target-specific interpretation, formatted as big-endian hex strings. A thread-id can also be a literal ‘-1’ to indicate all threads, or ‘0’ to pick any thread. 

[ref](https://sourceware.org/gdb/onlinedocs/gdb/Packets.html#thread_002did-syntax)

Reply:
I will handle this.